### PR TITLE
Media math

### PR DESCRIPTION
**Pull request by zan-kusterle**

### CODE REVIEW

File: src/stateless/DefaultListItem.vue [hunk 1]
```diff
@@ -185,7 +185,7 @@ export default {
         &__label {
             color: @gunpowder;
             &--selected { color: @royal-blue; }
-            &--disabled { color: @gray-blue; }
+            &--disabled { color: @very-light-gray; }
```
Comment (3buson) [inline src/stateless/DefaultListItem.vue]: Why this change? I'm a bit worried this might break sth since default list item is used on a lot of places

File: src/stateless/icon.vue [hunk 2]
```diff
@@ -77,6 +77,10 @@
             <path class="path" d="M8 0C3.6 0 0 3.6 0 8s3.6 8 8 8 8-3.6 8-8-3.6-8-8-8zm4 9H9v3H7V9H4V7h3V4h2v3h3v2z" fill-rule="nonzero"/>
         </svg>
 
+        <svg v-if="name === 'plus-v2'" viewBox="0 0 16 16" xmlns="http://www.w3.org/2000/svg">
```
Comment (3buson) [inline src/stateless/icon.vue]: Can we find a better name than `-v2`? It's not very descriptive. Maybe the original one can be `plus-round` and this one just `plus`?

File: src/stateless/input.vue [hunk 3]
```diff
@@ -568,6 +568,7 @@ export default {
         border: 0;
         outline: none;
         margin: 4px 0px;
+        transition: color @default-transition-time;
```
Comment (3buson) [inline src/stateless/input.vue]: Why was this change needed?

File: src/stateless/selectbox.vue [hunk 4]
```diff
@@ -69,6 +71,8 @@ export default {
             isOpen: false,
             focused: false,
             searchTextValue: '',
+            isSearchLoading: false,
```
Comment (3buson) [inline src/stateless/selectbox.vue]: Do we need this in data? Couldn't we store a promise and check if it is resolved?

File: src/stateless/selectbox.vue [hunk 5]
```diff
@@ -57,6 +57,8 @@ export default {
         warningText: { type: String, required: false, default: '' },
         options: { type: Array, required: true },
         value: { type: String, required: false, default: '' },
+        getOptions: { type: Function, required: false, default: null },
+        loadAsyncDebounce: { type: Number, default: 0 },
```
Comment (3buson) [inline src/stateless/selectbox.vue]: Does it make sense to make this configurable at this point? We would probably want it always anyway.

File: src/stateless/selectbox.vue [hunk 6]
```diff
@@ -125,6 +130,10 @@ export default {
             return this.disabled ? this.disabledText : ''
         },
         listItems () {
+            if (typeof this.getOptions === 'function'){
```
Comment (3buson) [inline src/stateless/selectbox.vue]: [minor] space before `{`

File: src/stateless/selectbox.vue [hunk 7]
```diff
@@ -153,6 +162,14 @@ export default {
             return options
         },
     },
+    watch: {
+        searchText () {
+            this.debouncedLoadAsyncOptions()
+        },
+    },
+    created () {
+        this.debouncedLoadAsyncOptions = debounce(this.loadAsyncOptions, this.loadAsyncDebounce)
```
Comment (3buson) [inline src/stateless/selectbox.vue]: Why isn't this a method?

File: src/stateless/selectbox.vue [hunk 8]
```diff
@@ -237,6 +255,16 @@ export default {
         setSearch: debounce(function (v) {
             this.searchTextValue = v
         }, 400),
+        loadAsyncOptions () {
+            if (typeof this.getOptions === 'function') {
```
Comment (3buson) [inline src/stateless/selectbox.vue]: When is getOptions not a function?

File: src/stateless/selectbox.vue [hunk 9]
```diff
@@ -237,6 +255,16 @@ export default {
...
+                this.isSearchLoading = true
+                this.getOptions(this.searchText || '')
+                    .then(options => {
+                        this.isSearchLoading = false
+                        this.asyncOptions = options
+                    }).catch(console.error)
```
Comment (3buson) [inline src/stateless/selectbox.vue]: We should probably catch error and log that error

File: src/stateless/selectbox.vue [hunk 10]
```diff
@@ -432,6 +460,10 @@ export default {
     .selectbox__label-text {
         color: @bluish-gray;
 
+        &--disabled {
```
Comment (3buson) [inline src/stateless/selectbox.vue]: Did we have the disabled state broken on master or why did we need to add this?

File: src/stateless/icon.vue (hunk 2 above)
Comment (jkosir) [reply src/stateless/icon.vue]: I'd rather not rename old ones, maybe ```cross```?

File: src/stateless/input.vue (hunk 3 above)
Comment (jkosir) [reply src/stateless/input.vue]: We don't have it in master now, but it should be there

File: src/stateless/selectbox.vue (hunk 4 above)
Comment (jkosir) [reply src/stateless/selectbox.vue]: Can't vue reactively handle promise state?

File: src/stateless/selectbox.vue (hunk 8 above)
Comment (jkosir) [reply src/stateless/selectbox.vue]: When it's not provided (could check if is null actually),

File: src/stateless/selectbox.vue (hunk 10 above)
Comment (jkosir) [reply src/stateless/selectbox.vue]: Yep. It's not used anywhere so we didn't notice it.

File: src/stateless/selectbox.vue (hunk 9 above)
Comment (jkosir) [reply src/stateless/selectbox.vue]: What do you mean? That's what it does.

File: src/stateless/icon.vue (hunk 2 above)
Comment (3buson) [reply src/stateless/icon.vue]: Yes, maybe it's safer not to do that now.
What about `plus-square`?

File: src/stateless/input.vue (hunk 3 above)
Comment (3buson) [reply src/stateless/input.vue]: So we broke master?

File: src/stateless/selectbox.vue (hunk 8 above)
Comment (3buson) [reply src/stateless/selectbox.vue]: Yes, let's check for null then

File: src/stateless/selectbox.vue (hunk 9 above)
Comment (3buson) [reply src/stateless/selectbox.vue]: :+1:

File: src/stateless/selectbox.vue (hunk 10 above)
Comment (3buson) [reply src/stateless/selectbox.vue]: :+1:

File: src/stateless/selectbox.vue (hunk 4 above)
Comment (3buson) [reply src/stateless/selectbox.vue]: We could probably have a computed then

File: src/stateless/icon.vue (hunk 2 above)
Comment (jkosir) [reply src/stateless/icon.vue]: 👍

File: src/stateless/DefaultListItem.vue (hunk 1 above)
Comment (jkosir) [reply src/stateless/DefaultListItem.vue]: Uh actually we don't need this now, this wouldn't change anything. Would have to add different colors on hover so it would make a difference.

Input is kinda strange now, for label text we have ```@gunpowder -> black``` on hover while the text entered in input is always black. This looks kinda strange next to selectboxes which have ```@gunpowder -> black``` both is selected or "Please select" state. We actually have it that way in designs, but should probably unify with other elements

Comment (jkosir) [reply src/stateless/DefaultListItem.vue]: commented on wrong issue...

File: src/stateless/input.vue (hunk 3 above)
Comment (jkosir) [reply src/stateless/input.vue]: Uh actually we don't need this now, this wouldn't change anything. Would have to add different colors on hover so it would make a difference.

Input is kinda strange now, for label text we have ```@gunpowder -> black``` on hover while the text entered in input is always black. This looks kinda strange next to selectboxes which have ```@gunpowder -> black``` both is selected or "Please select" state. We actually have it that way in designs, but should probably unify with other elements

File: src/stateless/DefaultListItem.vue (hunk 1 above)
Comment (jkosir) [reply src/stateless/DefaultListItem.vue]: This is for disabled selectbox state, we should probably override it in ```selectbox.vue```instead yeah

Comment (3buson) [reply src/stateless/DefaultListItem.vue]: Yes, this is a bit of a mess now with the whole 'inheritance' structure, we need to talk about how will we resolve this but for now an override would probably be better.

File: src/stateless/selectbox.vue (hunk 9 above)
Comment (3buson) [reply src/stateless/selectbox.vue]: Maybe it would make sense to also clear loading if search fails?
